WiinUPro/Windows/DevicePrefsWindow.xaml.cs

```diff
@@ -116,15 +166,22 @@ private void cancelBtn_Click(object sender, RoutedEventArgs e)
 
         private void btnDefaultProfile_Click(object sender, RoutedEventArgs e)
         {
-            Microsoft.Win32.OpenFileDialog dialog = new Microsoft.Win32.OpenFileDialog();
-            dialog.DefaultExt = ".wup";
-            dialog.Filter = App.PROFILE_FILTER;
+            FindProfile(defaultProfile);
+        }
 
-            bool? didFind = dialog.ShowDialog();
+        private void btnExtProfile_Click(object sender, RoutedEventArgs e)
+        {
+            FindProfile(extProfile);
+            _modifiedPrefs.extensionProfiles[comboExtProfile.SelectedIndex] = extProfile.Text;
+        }
 
-            if (didFind == true && dialog.CheckFileExists)
+        private void comboExtProfSelect_SelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+            if (selectionChanged)
             {
-                defaultProfile.Text = dialog.FileName;
+                _modifiedPrefs.extensionProfiles[lastIndex] = extProfile.Text;

```

Comment:
So if index is 0, No Extension, that is pretty much default yeah?

Reply:
Pretty much, extension-wise. Unlike defaultProfile however, the "No Extension" profile won't be loaded unless there's no extension attached. Hence, one may for example set defaultProfile to a profile for holding the Wiimote with one hand and "No Extension" to a profile specifically made for holding and using the Wiimote like a NES controller. In this case defaultProfile would be loaded when using a Nunchuk or a guitar (unless those two have their own separate profile paths specified), whereas if no extension is connected the "No Extension" profile would be loaded.

Reply:
One extension specific profile path (or the default profile) will always be _technically_ unnecessary since one can just bind all the other profile paths to a specific profile, but that's not really ideal for the end user.

Reply:
I see. If we made it such that if there was no profile specified for an extension it would just load the 0th indexed profile if it exists, that would consolidate them. Though I suppose I can see reasons to keep them separate. Like if the default contains different configurations for mapping to a 360 controller and only one specific No Extension mapping.

I'm also wondering if a dictionary might serve this better and if we could have the UI be less static; but I'm just thinking _out loud_.